### PR TITLE
Late escape RSS block

### PR DESCRIPTION
**Pull request by getdave**
## Description
This is not a security problem. This PR simply moves [escaping of all PHP output to be as "late" as possible](https://developer.wordpress.org/themes/theme-security/data-sanitization-escaping/#:~:text=esc_textarea(%20%24text%20)%3B%20%3F%3E%3C/textarea%3E-,Tip%3A,Output%20escaping%20should%20occur%20as%20late%20as%20possible.,-Top%20%E2%86%91). This means we avoid escaping variables until they are output in the HTML markup.

This is a WP Core best practice.

## How has this been tested?
- add both blocks
- check functionality is "as was"
- check all tests pass

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/block-library/src/rss/index.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ function render_block_core_rss( $attributes ) {
 		$excerpt = '';
 		if ( $attributes['displayExcerpt'] ) {
 			$excerpt = html_entity_decode( $item->get_description(), ENT_QUOTES, get_option( 'blog_charset' ) );
-			$excerpt = esc_attr( wp_trim_words( $excerpt, $attributes['excerptLength'], ' [&hellip;]' ) );
```
Comment (getdave) [inline packages/block-library/src/rss/index.php:69]: I noticed this is being escaped early with `esc_attr` and then again later with `esc_html`. I assume the `esc_attr` is an error but would appreciate a confidence check @hellofromtonya

File: packages/block-library/src/rss/index.php [hunk 2]
```diff
@@ -33,9 +33,9 @@ function render_block_core_rss( $attributes ) {
 		$link = $item->get_link();
 		$link = esc_url( $link );
```
Comment (Mamaduka) [inline packages/block-library/src/rss/index.php]: I think we don't need this here anymore.

File: packages/block-library/src/rss/index.php (hunk 1 above)
Comment (getdave) [reply packages/block-library/src/rss/index.php:69]: Or anyone really 😄

Comment (Mamaduka) [reply packages/block-library/src/rss/index.php:69]: Looks good—no need to double escape here. Btw `esc_html` and `esc_attr` are the same under the hood.

Comment (getdave) [reply packages/block-library/src/rss/index.php:69]: Really!? Why then do we have both? 🤔

Comment (Mamaduka) [reply packages/block-library/src/rss/index.php:69]: Probably for backward compatibility, plus it's easier to remember escape function names this way.

File: packages/block-library/src/rss/index.php (hunk 2 above)
Comment (getdave) [reply packages/block-library/src/rss/index.php]: Good spot. Fixed!

**Review by Mamaduka — APPROVED** [review]
Thanks, Dave. Changes look good to me.

File: packages/block-library/src/rss/index.php [hunk 3]
```diff
@@ -16,11 +16,11 @@ function render_block_core_rss( $attributes ) {
 	$rss = fetch_feed( $attributes['feedURL'] );
 
 	if ( is_wp_error( $rss ) ) {
-		return '<div class="components-placeholder"><div class="notice notice-error"><strong>' . __( 'RSS Error:' ) . '</strong> ' . $rss->get_error_message() . '</div></div>';
+		return '<div class="components-placeholder"><div class="notice notice-error"><strong>' . esc_html__( 'RSS Error:' ) . '</strong> ' . esc_html( $rss->get_error_message() ) . '</div></div>';
 	}
 
 	if ( ! $rss->get_item_quantity() ) {
-		return '<div class="components-placeholder"><div class="notice notice-error">' . __( 'An error has occurred, which probably means the feed is down. Try again later.' ) . '</div></div>';
+		return '<div class="components-placeholder"><div class="notice notice-error">' . esc_html__( 'An error has occurred, which probably means the feed is down. Try again later.' ) . '</div></div>';
```
Comment (ntsekouras) [inline packages/block-library/src/rss/index.php:23]: Why would we need to escape this? 🤔  It's a hardcoded string.

Comment (getdave) [reply packages/block-library/src/rss/index.php:23]: I've been discussing this with the Core Security folks. Ultimately we've agreed that in general WP Core accepts that translations are safe to output without escaping. In previous roles when I've done security testing and auditing I've always just escaped everything zealously including translations so I was following the same line of thinking.

I'm on a learning curve here about the expectations in WP Core regarding this some of which is not documented.

If you'd like I'm happy to undo this change but it's probably pretty low impact.

**Pull request merged**